backend/controllers/userController.js

```diff
@@ -4,21 +4,44 @@ module.exports = {
 
     search: async function(req, res) {
 
-        // 활성화 상태의 유저만 조회
-        let query = Object.assign(req.query, { status: 'avail' });
-        let projection = {
-            name: true, rank: true, title: true,
-            group: true, email: true, tel: true
+        const keys = Object.keys(req.query);
+        const valids = ['group', 'name'];
+            
+         // Only allowed fields are Searchable
+        for(let key of keys) {
+            if(!valids.includes(key))
+                throw new BusinessError(`${key} is not allowed param`);
+        }
+
+
+        const projection = {
+            _id: true, name: true, rank: true, status: true,
+            group: true, email: true, tel: true, lastLogin: true
         };
 
         try {
-            const result = await userService.search(query, projection);
+            const result = await userService.find(req.query, projection);
             res.status(200).send(result);
         } catch(err) {
             res.status(err.status).send(err.message);
         }
     },
 
+    searchDetail: async function(req, res) {
+
+        const projection = {
+            _id: true,serviceNumber:true, name: true, rank: true, title:true,
+            status: true, group: true, email: true, tel: true, lastLogin: true,
+            lastLogin: true, firseLogin:true, bookmarks: true, subscriptions: true
+        };
+        try {
+            const result = await userService.findOne({_id: req.params.id}, projection);
+            res.status(201).send(result);   // 201 Created
```

Comment:
Status code `201` is using for when resource has created.
Status code `200` is cut out for this function.

Reply:
As @134130 said, `200` should be used in this case.